tests/pipelines/test_train_step.py

```diff
@@ -60,7 +60,14 @@ def setup_train_dataset(pipeline_root: Path):
 
 
 # Sets up the constructed TrainStep instance
-def setup_train_step(pipeline_root: Path, use_tuning: bool, with_hardcoded_params: bool = True):
+def setup_train_step(
```

Comment:
Can we split this function into `setup_train_step_with_tuning` and `setup_train_step_with_automl`? They are not very related. The function has become difficult to reasonable with so many knobs.

Reply:
done!